include/sparrow/arrow_interface/arrow_array.hpp

```diff
@@ -0,0 +1,193 @@
+     * @param n_buffers The number of physical buffers backing this array. The number of buffers is a
+     *                  function of the data type, as described in the Columnar format specification, except
+     *                  for the the binary or utf-8 view type, which has one additional buffer compared to
+     *                  the Columnar format specification (see Binary view arrays). Must be 0 or positive.
+     * @param buffers Vector of sparrow::buffer
+     * @param children Vector of arrow_array_shared_ptr representing the children of the ArrowArray.
+     * @param dictionary arrow_array_shared_ptr or nullptr.
+     * @return The created ArrowArray.
+     */
+    template <class B, class C, class D>
+        requires std::constructible_from<arrow_array_private_data::BufferType, B>
+                 && std::constructible_from<arrow_array_private_data::ChildrenType, C>
+                 && std::constructible_from<arrow_array_private_data::DictionaryType, D>
+    arrow_array_unique_ptr make_arrow_array_unique_ptr(
+        int64_t length,
+        int64_t null_count,
+        int64_t offset,
+        int64_t n_buffers,
+        B buffers,
+        int64_t n_children,
+        C children,
+        D dictionary
+    );
+
+    /**
+     * @brief Creates a unique pointer to an Arrow array.
+     *
+     * This function creates a unique pointer to an Arrow array with the specified parameters.
+     *
+     * @tparam B Value, reference or rvalue of std::vector<sparrow::buffer<uint8_t>>
+     * @tparam C Value, reference or rvalue of std::vector<arrow_array_shared_ptr>
+     * @tparam D Value, reference or rvalue of arrow_array_shared_ptr
+     * @param length The logical length of the array (i.e. its number of items). Must be 0 or positive.
+     * @param null_count The number of null items in the array. May be -1 if not yet computed. Must be 0 or
+     * positive otherwise.
+     * @param offset The logical offset inside the array (i.e. the number of items from the physical start of
+     *               the buffers). Must be 0 or positive.
+     * @param buffers Vector of sparrow::buffer<uint8_t>.
+     * @param children Vector of arrow_array_shared_ptr representing the children of the ArrowArray.
+        * @param dictionary arrow_array_shared_ptr or nullptr.mp
+     * @return The created ArrowArray.
+     */
+    template <class B, class C, class D>
+        requires std::constructible_from<arrow_array_private_data::BufferType, B>
+                 && std::constructible_from<arrow_array_private_data::ChildrenType, C>
+                 && std::constructible_from<arrow_array_private_data::DictionaryType, D>
+    arrow_array_unique_ptr
+    make_arrow_array_unique_ptr(int64_t length, int64_t null_count, int64_t offset, B buffers, C children, D dictionary);
+
+    /**
+     * Creates a unique pointer to an Arrow array with default values.
+     *
+     * This function creates a unique pointer to an Arrow array with default values.
```

Comment:
Redundant with the first line. 
Instead, maybe clarify what default values should be expected?

Reply:
Ok done